test/MeshKernelNETTest/Api/MeshKernelTest.cs

```diff
@@ -2493,5 +2493,56 @@ public void Mesh2dMakeGlobaThroughApi()
                 }
             }
         }
+
+
+        [Test]
+        public void Mesh2dUndoRedoThroughApi()
+        {
+            // Setup
+            using (DisposableMesh2D mesh = CreateMesh2D(4, 4, 100, 200))
+            using (var api = new MeshKernelApi())
+            {
+                var id = 0;
+                var mesh2d = new DisposableMesh2D();
+                try
+                {
+                    int numberOfVerticesBefore = mesh.NumNodes;
+                    id = api.AllocateState(0);
+
+                    Assert.AreEqual(0, api.Mesh2dSet(id, mesh));
+
+                    // Do
+                    Assert.AreEqual(0, api.Mesh2dDeleteNode(id, 0));

```

Comment:
Consider just calling the delete-node without asserting its output, or getting the intermediate data. That behavior is being tested elsewhere. The subject of this test is the undo of the delete-node calls.

It _would_ be interesting to get the mesh after the delete-node calls and show that nodes 0 and 6 are indeed invalid.